src/tokenizer/mod.rs

```diff
@@ -1423,4 +1534,20 @@ mod test {
         option_push(&mut s, 'x');
         assert_eq!(s, Some("yx".to_tendril()));
     }
+
+    #[test]
+    fn check_lines() {
+
+        let opts = TokenizerOpts{exact_errors: false, discard_bom: true, profile: false,
+            initial_state: None, last_start_tag_name: None,};
+
+        let vector = vec![StrTendril::from("<a>\r"), StrTendril::from("<b>\r"),
+            StrTendril::from("</b>\r"), StrTendril::from("</a>\r")];
+        let expected = vec![1, 2, 2, 3, 3, 4, 4, 5, 5];
```

Comment:
We'll need to update this to contain pairs of tokens and the corresponding line number.